File: rent-now-ui/src/app/services/booking/BookingService.ts
import {BookingRQ} from "../../data/model/rq/BookingRQ";
import {Observable} from "rxjs";
import {BookingRS} from "../../data/model/rs/BookingRS";
import {Page} from "../../data/model/common/Page";

export interface BookingService {

  addBooking(data: BookingRQ): Observable<BookingRS>;

  getBooking(bookingId: number): Observable<BookingRS>;

  cancelBooking(bookingId: number): Observable<String>;

  getBookingsByUser(uuid: string, page: number, size: number, sort: string): Observable<Page<BookingRS>>;
}

File: rent-now-backend/src/main/java/pl/holowinska/rentnowbackend/services/BookingService.java
package pl.holowinska.rentnowbackend.services;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import pl.holowinska.rentnowbackend.exceptions.AccommodationNotFoundException;
import pl.holowinska.rentnowbackend.exceptions.BookingConflictException;
import pl.holowinska.rentnowbackend.exceptions.BookingNotFoundException;
import pl.holowinska.rentnowbackend.model.rq.BookingRQ;
import pl.holowinska.rentnowbackend.model.rs.BookingRS;

import java.time.LocalDate;
import java.util.List;

public interface BookingService {

    BookingRS addBooking(BookingRQ bookingRQ) throws AccommodationNotFoundException, BookingConflictException;

    void deleteBooking(Long bookingId) throws BookingNotFoundException;

    BookingRS getBooking(Long bookingId) throws BookingNotFoundException;

    Page<BookingRS> getBookingListByUser(String uuid, Pageable pageable);

    Page<BookingRS> getBookingListByAccommodation(String accommodationId, Pageable pageable);

    List<LocalDate> getBookedStartDatesByAccommodation(String accommodationId);

    List<LocalDate> getBookedEndDatesByAccommodation(String accommodationId);
}
